### PR TITLE
fix issue #282 to delete multiple resources

### PR DESCRIPTION
**Pull request by mphammer**

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/355#" title="Author self-approved">mphammer</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/synopsysctl/cmd_delete.go [hunk 1]
```diff
@@ -40,75 +40,69 @@ var deleteCmd = &cobra.Command{
...
-		// Delete Blackduck with Client
-		err := blackduckClient.SynopsysV1().Blackducks(blackduckNamespace).Delete(blackduckNamespace, &metav1.DeleteOptions{})
-		if err != nil {
-			log.Errorf("error deleting the Blackduck: '%s'", err)
-			return nil
+		for _, blackduckNamespace := range args {
+			log.Infof("Deleting BlackDuck %s...", blackduckNamespace)
+			err := blackduckClient.SynopsysV1().Blackducks(blackduckNamespace).Delete(blackduckNamespace, &metav1.DeleteOptions{})
+			if err != nil {
+				log.Errorf("error deleting the Blackduck: '%s'", err)
```
Comment (rrati) [inline pkg/synopsysctl/cmd_delete.go]: Provide the name of the blackduck with the error as well.  "error deleting black duck %s": %+v", name, err)

File: pkg/synopsysctl/cmd_delete.go [hunk 2]
```diff
@@ -40,75 +40,69 @@ var deleteCmd = &cobra.Command{
...
-		// Delete OpsSight with Client
-		err := opssightClient.SynopsysV1().OpsSights(opsSightNamespace).Delete(opsSightNamespace, &metav1.DeleteOptions{})
-		if err != nil {
-			log.Errorf("error deleting the OpsSight: '%s'", err)
-			return nil
+		for _, opsSightNamespace := range args {
+			log.Infof("Deleting OpsSight %s...", opsSightNamespace)
+			err := opssightClient.SynopsysV1().OpsSights(opsSightNamespace).Delete(opsSightNamespace, &metav1.DeleteOptions{})
+			if err != nil {
+				log.Errorf("error deleting the OpsSight: '%s'", err)
```
Comment (rrati) [inline pkg/synopsysctl/cmd_delete.go]: Same comment as blackducks

File: pkg/synopsysctl/cmd_delete.go [hunk 3]
```diff
@@ -40,75 +40,69 @@ var deleteCmd = &cobra.Command{
...
-		// Delete Alert with Client
-		err := alertClient.SynopsysV1().Alerts(alertNamespace).Delete(alertNamespace, &metav1.DeleteOptions{})
-		if err != nil {
-			log.Errorf("error deleting the Alert: %s", err)
-			return nil
+		for _, alertNamespace := range args {
+			log.Infof("Deleting Alert %s...", alertNamespace)
+			err := alertClient.SynopsysV1().Alerts(alertNamespace).Delete(alertNamespace, &metav1.DeleteOptions{})
+			if err != nil {
+				log.Errorf("error deleting the Alert: %s", err)
```
Comment (rrati) [inline pkg/synopsysctl/cmd_delete.go]: Same here

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/355#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/355#pullrequestreview-238042683" title="Approved">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/synopsysctl/cmd_delete.go [hunk 4]
```diff
@@ -40,75 +40,69 @@ var deleteCmd = &cobra.Command{
...
 // deleteBlackduckCmd deletes a Blackduck from the cluster
 var deleteBlackduckCmd = &cobra.Command{
-	Use:   "blackduck NAMESPACE",
-	Short: "Delete a Blackduck",
+	Use:   "blackduck NAME...",
+	Short: "Delete one or many Black Ducks",
 	Args: func(cmd *cobra.Command, args []string) error {
-		if len(args) != 1 {
-			return fmt.Errorf("this command takes 1 argument")
+		if len(args) <= 0 {
```
Comment (msenmurugan) [inline pkg/synopsysctl/cmd_delete.go]: == 0 should be good.

File: pkg/synopsysctl/cmd_delete.go [hunk 5]
```diff
@@ -40,75 +40,69 @@ var deleteCmd = &cobra.Command{
...
 // deleteOpsSightCmd deletes an OpsSight from the cluster
 var deleteOpsSightCmd = &cobra.Command{
-	Use:   "opssight NAMESPACE",
-	Short: "Delete an OpsSight",
+	Use:   "opssight NAME...",
+	Short: "Delete one or many OpsSights",
 	Args: func(cmd *cobra.Command, args []string) error {
-		if len(args) != 1 {
-			return fmt.Errorf("this command takes 1 argument")
+		if len(args) <= 0 {
```
Comment (msenmurugan) [inline pkg/synopsysctl/cmd_delete.go]: same comment as blackduck

File: pkg/synopsysctl/cmd_delete.go [hunk 6]
```diff
@@ -40,75 +40,69 @@ var deleteCmd = &cobra.Command{
...
 // deleteAlertCmd deletes an Alert from the cluster
 var deleteAlertCmd = &cobra.Command{
-	Use:   "alert NAMESPACE",
-	Short: "Delete an Alert",
+	Use:   "alert NAME...",
+	Short: "Delete one or many Alerts",
 	Args: func(cmd *cobra.Command, args []string) error {
-		if len(args) != 1 {
-			return fmt.Errorf("this command takes 1 argument")
+		if len(args) <= 0 {
```
Comment (msenmurugan) [inline pkg/synopsysctl/cmd_delete.go]: same comment as blackduck

**Pull request merged**